rust/lance-index/src/vector/pq.rs

```diff
@@ -109,6 +109,15 @@ impl ProductQuantizer {
         let num_sub_vectors = self.num_sub_vectors;
         let dim = self.dimension;
         let num_bits = self.num_bits;
+        if num_bits == 4 && num_sub_vectors % 2 != 0 {
```

Comment:
nit: can we pad instead of requireing this? not a big deal as I think it's almost always a multiple of 2